filebeat/fileset/config.go

```diff
@@ -0,0 +1,17 @@
+package fileset
+
+// ModuleConfig contains the configuration file options for a module
+type ModuleConfig struct {
+	Module   string                    `config:"module"     validate:"required"`
+	Enabled  *bool                     `config:"enabled"`
+	Filesets map[string]*FilesetConfig `config:"filesets"`
```

Comment:
I would vote for the additional complexity to replace filesets from the config and be consistent with metricbeat. This also make the command line options look more natural as there it is already without `filesets`. The nice part about having *common.Config that it provides already an `Enabled()` function, so not even unpacking is needed.

I didn't check the code details on this yet, but one solution is to work in general with *common.Config objects which are passed around and unpacking happens locally. So also each fileset will unpack its own config options